Drop switch from Redis unlink to delete

michaldo opened the pull request:
Drop support for Redis 3, which is not maintained since 2018

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/main/CONTRIBUTING.adoc).
-->

artembilan reviewed (changes requested):
After discussion with Spring Data team, we agreed that fully dropping `DEL` support is fair game from now on.
Please, consider to refine the logic just deal with `UNLINK`.
Also, take a look into `RedisQueueInboundGateway`, `RedisQueueMessageDrivenEndpoint` and `RedisMessageStore` where we still use `DEL`.

Thanks

michaldo commented:
I decided to remove switch from unlink to dev
About places where del is used, I will take a look later

artembilan commented:
> I will take a look later

Thank you!

Let's us know, please, if you are going to that as a part of this PR or so!
In the end we would need to mention in the `whats-new.adoc` such a change in Redis module.
Also, the `redis.adoc` need to be revised if we still mention `DEL` somewhere there and probably an algorithm you've just dropped.

artembilan reviewed (changes requested):
Also, just noticed:
```
Signed-off-by: michaldo <michaldo@github.com>
```

Please, use your legal name instead according to DCO requirements.
You can configure your Git client to produce a proper name for you into those commits.

Commit:
Carefully switch from Redis unlink to delete

More performant `unlink` is not available on Redis 3.

Implementation is aware that, however not careful exception handling may switch to less performant `delete` on connection exception or similar case. The switch is permanent.

Maybe better choice is drop support for Redis 3, which is not maintained since 2018

Signed-off-by: Michal Domagala <michaldo@github.com>

Commit:
Drop fallback from Redis unlink to delete

Redis 3, target of fallback, is not maintained since 2018

Signed-off-by: Michal Domagala <michaldo@github.com>

Commit:
Drop fallback from Redis unlink to delete

Remove code remainings

Signed-off-by: Michal Domagala <michaldo@github.com>

Commit:
Checkstyle

Signed-off-by: Michal Domagala <michaldo@github.com>

Commit:
Checkstyle

Signed-off-by: Michal Domagala <michaldo@github.com>

artembilan reviewed (changes requested):
Please, add your name into the `@author` list of the affected class.
Also, give us your thought about other classes in Redis module which still use `DEL`.

Thank you again!

Commit:
Author

Signed-off-by: Michal Domagala <michaldo@github.com>

Commit:
Same cleanup in RedisMessageStore

Signed-off-by: Michal Domagala <michaldo@github.com>

michaldo commented:
I add myself as one of `@author`'s
Regarding other classes:

- RedisMessageStore: is it same case switch from unlink -> delete and I removed it also

- RedisQueueInboundGateway, RedisQueueMessageDrivenEndpoint: they use delete command within JMX `@ManagedOperation` command. I feel that JMX action looks like admin action and I think If were someone who executes admin action I prefer to be sure that when action is completed everything is done. It mean I prefer synchronous delete over asynchronous unlink
